src/group/mls_group/new_from_welcome.rs

```diff
@@ -121,7 +121,7 @@ impl MlsGroup {
                 tree.own_node_index(),
                 NodeIndex::from(group_info.signer_index()),
             );
-            let common_path = treemath::direct_path_root(common_ancestor_index, tree.leaf_count())
+            let common_path = treemath::dirpath_long(common_ancestor_index, tree.leaf_count())
```

Comment:
The common path now also includes the leaf with that change. That seems wrong conceptually, since the own leaf node cannot be on the same path as any other node.

Reply:
The common path is computed from the `comon_ancestor_index`, not from the leaf. So it's meant to include that node as well, which I think is correct. I'd like to keep this here for now. I'll see if I can extract it into the new binary tree interface.

Reply:
The part that I don't understand is that you now have a `common_path` that is longer than previously. Why does that not lead to errors?

While you are here, could you also get rid of the `expect()` and propagate the error? Or convert it to an `unwrap()` with a proper explanation.

Reply:
It's not necessarily longer. If I understand correctly, it's only longer if the common ancestor is not the root. And since the bug essentially occurs when that is the case, I think that it now works correctly and it previously didn't (or can't have).